### DISCUSSION TITLE
But what about get/post/and other methods in server api?

### DISCUSSION THREAD

**Discussion by n4an** [nuxt/nuxt#17448] [Questions]
```
Example: Simple API route

// routes/test.ts
export default eventHandler(() => 'Hello World!')
Example: API route with params

// routes/hello/[name].ts
export default eventHandler(event => `Hello ${event.context.params.name}!`)
Example: Catch all page

// routes/[...].ts
export default eventHandler(event => `Default page`)
```

That routes will work for all http methods?
Is need to play with something like `if (event.context.method == 'get') {res.send('ok')} else {throw new Error }`?

**Comment by Diizzayy** [comment]
> That routes will work for all http methods?

@n4an Indeed, the routes by default will work for all HTTP Methods.

If you wish to explicitly assert a specify HTTP Method:

- You can leverage `assertMethod` as seen below, 
```ts
export default defineEventHandler(async (event) => {
    assertMethod(event, 'GET');

    return 'EXPLICIT GET REQUEST'
})
```

- Or make use of [h3 method matching](http://v3.nuxtjs.org/guide/features/server-routes#matching-http-method)

**Reply by n4an** [reply]
`Or make use of [h3 method matching](http://v3.nuxtjs.org/guide/features/server-routes#matching-http-method)`
That is more interesting. I hope there is such a thing.

**Reply by durbanitas** [reply]
@Diizzayy is there a difference between `eventHandler()` and `defineEventHandler()`? 

Can't find any description.

**Reply by Diizzayy** [reply]
@durbanitas none whatsoever, you can use whichever you prefer `defineEventHandler` is just to maintain the `defineXY` style used throughout Nuxt3

**Comment by novvac** [comment]
https://v3.nuxtjs.org/guide/features/server-routes#matching-http-method
